client/me/account/main.jsx

```diff
@@ -84,25 +82,58 @@ const Account = React.createClass( {
 		debug( this.constructor.displayName + ' component is unmounting.' );
 	},
 
-	updateLanguage() {
-		let valueLink = this.valueLink( 'language' );
+	getUserSetting( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
+	},
 
-		valueLink.requestChange = ( value ) => {
-			const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
+	updateUserSetting( settingName, value ) {
+		this.props.userSettings.updateSetting( settingName, value );
+	},
 
-			this.props.userSettings.updateSetting( 'language', value );
-			if ( value !== originalLanguage ) {
-				this.setState( { redirect: '/me/account' } );
-			} else {
-				this.setState( { redirect: false } );
-			}
-		};
+	updateUserSettingInput( settingName ) {
+		return event => this.updateUserSetting( settingName, event.target.value );
+	},
 
-		return valueLink;
+	updateUserSettingCheckbox( settingName ) {
+		return event => this.updateUserSetting( settingName, event.target.checked );
+	},
+
+	updateLanguage( event ) {
+		const { value } = event.target;
+		const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
+
+		this.updateUserSetting( 'language', value );
+		if ( value !== originalLanguage ) {
+			this.setState( { redirect: '/me/account' } );
+		} else {
+			this.setState( { redirect: false } );
+		}
```

Comment:
Yes, it is used by the `formBase` mixin: https://github.com/Automattic/wp-calypso/blob/master/client/me/form-base/index.js#L81-L86

I'd like to get rid of the redirect eventually and apply the new language without reload, but that's another story (and PR).